pennylane/qnode.py

```diff
@@ -616,6 +619,40 @@ def __call__(self, *args, **kwargs):
         )
         self._tape_cached = using_custom_cache and self.tape.hash in cache
 
+        if qml.query_return():
+            res = qml.execute_new(
+                [self.tape],
+                device=self.device,
+                gradient_fn=self.gradient_fn,
+                interface=self.interface,
+                gradient_kwargs=self.gradient_kwargs,
+                override_shots=override_shots,
+                **self.execute_kwargs,
+            )
+
+            if not isinstance(self._qfunc_output, tuple) and not isinstance(
+                self._qfunc_output, qml.measurements.MeasurementProcess
+            ):

```

Comment:
```suggestion
            if not isinstance(self._qfunc_output, (tuple, qml.measurements.MeasurementProcess)):

```

Reply:
1) Yes it is correct, if it we have tuples we already have the desired results.
2) Yes also correct, the only case that is possible for single measurement is the one where we want to use a list. And I added a special case for it.